### ISSUE TITLE
GraphQL endpoint

### ISSUE THREAD

**Issue by baelter** [rabbitmq/rabbitmq-management#471]
How about adding a GraphQL endpoint ro the http  api. Possibly usling https://github.com/graphql-erlang/graphql

**Comment by michaelklishin** [comment]
@baelter that's a very brief description. What should it cover? What would the benefits be? We have dozens of endpoints, some of them do not assume "querying".

**Comment by baelter** [comment]
Sorry for the short explanation. So a bit of back story:

I have been thinking about a new UI for the http endpoints. Basically #113 
While researching for that I've been reading up on GraphQL and wanted to start a discussion around using it in the the management plugin.

The benefits from using GraphQL instead of RESTish take I would like to utilize would be

**No over fetching**
The http api has ?columns, paging and ?name filtering, but a graphql endpoint could provide more filtering options.

**No under fetching**
No 1+N request problems. Like:
Get queues for this vhost? 
Get queue details for each queue.

So basically being able to ask what exactly the data you need.
